src/odemis/driver/orsay.py

```diff
@@ -0,0 +1,1346 @@
+        """
+
+        model.HwComponent.__init__(self, name, role, parent=parent, **kwargs)
+
+        self._blevel = None
+
+        self.level = model.FloatContinuous(1.0, range=(0.0, 1.0), readonly=True, unit="")
+
+        self.on_connect()
+
+    def on_connect(self):
+        """
+        Defines direct pointers to server components and connects parameter callbacks for the Orsay server.
+        Needs to be called after connection and reconnection to the server.
+        """
+        self._blevel = self.parent.datamodel.HybridPlatform.UPS.UPScontroller.BatteryLevel
+
+        self._blevel.Subscribe(self._updateLevel)
+
+        self.update_VAs()
+
+    def update_VAs(self):
+        """
+        Update the VA's. Should be called after reconnection to the server
+        """
+        self._updateLevel()
+
+    def _updateLevel(self, parameter=None, attributeName="Actual"):
+        """
+        parameter (Orsay Parameter): the parameter on the Orsay server to use to update the VA
+        attributeName (str): the name of the attribute of parameter which was changed
+
+        Reads the battery level of the UPS from the Orsay server and saves it in the level VA
+        """
+        if parameter is None:
+            parameter = self._blevel
+        if parameter is not self._blevel:
+            raise ValueError("Incorrect parameter passed to _updateLevel. Parameter should be "
+                             "datamodel.HybridPlatform.UPS.UPScontroller.BatteryLevel")
+        if attributeName != "Actual":
+            return
+        currentLevel = float(parameter.Actual)
+        self.level._set_value(currentLevel / 100, force_write=True)
+
+    def terminate(self):
+        """
+        Called when Odemis is closed
+        """
+        self._blevel.Unsubscribe(self._updateLevel)
+        _blevel = None
+
+
+class GIS(model.Actuator):
+    """
+    This represents the GIS from Orsay Physics
+    """
+
+    def __init__(self, name, role, parent, **kwargs):
+        """
+        Has axes:
+        • "operational": unit is "None", choices is (True, False)
+
+        Defines the following VA's and links them to the callbacks from the Orsay server:
+        • position (VA, read-only, value is {"operational" : _positionPar.Actual})
+        • gasOn (BooleanVA, set to True to open/start the gas flow and False to close/stop the gas flow)
+        """
+        axes = {"operational": model.Axis(unit=None, choices={True, False})}
+
+        model.Actuator.__init__(self, name, role, parent=parent, axes=axes, **kwargs)
+
+        self._gis = None
+        self._errorPar = None
+        self._positionPar = None
+        self._reservoirPar = None
+
+        self._operationalReached = threading.Event()
+        self._operationalReached.set()
+
+        self.position = model.VigilantAttribute({"operational": False}, readonly=True)
+        self.gasOn = model.BooleanVA(False, setter=self._setGasOn)
```

Comment:
I'm not super found of the name... but I have hard time finding something better ;-)
I'd like something that sounds more like a setting. Maybe "opening" (but you'd want to put a size), or "opened"....